service/catalog.go

```diff
@@ -68,14 +101,36 @@ func createCatalog(w http.ResponseWriter, r *http.Request, envId string) (int, e
 		return http.StatusBadRequest, errors.New("Missing field 'url'")
 	}
 
+	if isDuplicateName(catalogModel) {
+		return http.StatusUnprocessableEntity, errors.New("Cannot add catalog with duplicate field `name`")
+	}
+
 	if err := db.Create(catalogModel).Error; err != nil {
 		return http.StatusBadRequest, err
 	}
 
-	apiContext.Write(catalogResource(catalogModel.Catalog, apiContext))
+	apiContext.Write(catalogResource(catalogModel.Catalog, apiContext, ""))
 	return 0, nil
 }
 
+func catalogExists(catalogModel *model.CatalogModel, envId string) bool {
+	query := `
+	SELECT *
+	FROM catalog
+	WHERE name = ?
+	AND environment_id = ?`
+
+	catalog := []model.CatalogModel{}
+
+	db.Raw(query, catalogModel.Name, envId).Find(&catalog)
+
+	if len(catalog) > 0 {
+		return false
+	}
+
+	return true
+}
+
 func updateCatalog(w http.ResponseWriter, r *http.Request, envId string) (int, error) {
```

Comment:
Am generally worried about the ability to update a catalog name, which is effectively its ID. Would like to discuss with you and @vincent99